orchestrator/src/worker/event_handler/triggers/aggregator_batching.rs

```diff
@@ -0,0 +1,138 @@
+
+            // If there are no blocks to process, return early
+            if start_block > end_block {
+                debug!("No Aggregator blocks to process: start_block ({}) > end_block ({})", start_block, end_block);
+                return Ok(());
+            }
+
+            info!("Processing Aggregator batches for blocks {} to {}", start_block, end_block);
+
+            let mut state = state_handler.load_batch_state().await?;
+
+            for block_num in start_block..=end_block {
+                match batching_handler.include_block(block_num, state).await? {
+                    BlockProcessingResult::Accumulated(updated_state) => {
+                        state = AggregatorState::NonEmpty(updated_state);
+                    }
+                    BlockProcessingResult::BatchCompleted { completed_state, new_state } => {
+                        state_handler.save_batch_state(&completed_state).await?;
+                        state = new_state;
+                    }
+                    BlockProcessingResult::NotBatched(current_state) => {
+                        state = current_state;
+                        // Since this block wasn't able to get batches, we shouldn't move forward
+                        break;
+                    }
+                }
+            }
+
+            match state {
+                AggregatorState::Empty(_) => {}
+                AggregatorState::NonEmpty(state) => {
+                    state_handler.save_batch_state(&state).await?;
+                }
+            }
+
+            Ok(())
+        }
+        .await;
+
+        // Always release the lock, regardless of whether work succeeded or failed
+        if let Err(e) = config.lock().release_lock("AggregatorBatchingWorker", None).await {
+            error!("Failed to release AggregatorBatchingWorker lock: {}", e);
+            // If work succeeded but lock release failed, return the lock release error
+            if result.is_ok() {
+                return Err(e.into());
+            }
+            // If work failed, we still want to return the original work error
+        }
+
+        result
+    }
+}
+
+impl AggregatorBatchingTrigger {
+    async fn calculate_range(&self, config: &Arc<Config>) -> Result<(u64, u64), JobError> {
+        // Get the latest aggregator batch from the DB
+        let latest_batch = config.database().get_latest_aggregator_batch().await?;
+
+        // Getting the latest block numbers for aggregator and snos batches from DB
+        let last_block_in_batches = latest_batch.map_or(-1, |batch| batch.end_block as i64);
+
+        // Getting the latest block number from the sequencer
+        let provider = config.madara_rpc_client();
+        let last_block_in_provider =
+            provider.block_number().await.map_err(|e| JobError::ProviderError(e.to_string()))?;
+
+        // Calculating the last block number that needs to be assigned to a batch
+        let last_block_to_assign_batch = config
```

Comment:
```suggestion
        let end_block = config
```

this shouldn't conflict w/ the next `let end_block=` because rust allows multiple declarations
and might be more readable that we're overriding the end_block based on `max_blocks_to_process_at_once`

Reply:
gotcha